File: chain-sdk/clojure/src/on_chain_api/specs/swap_v3_dto.clj
(ns on-chain-api.specs.swap-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.e-trade-aggressive-side :refer :all]
            )
  (:import (java.io File)))


(def swap-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :transaction) string?
   (ds/opt :timestamp) inst?
   (ds/opt :pool) string?
   (ds/opt :token_0) string?
   (ds/opt :token_1) string?
   (ds/opt :sender) string?
   (ds/opt :recipient) string?
   (ds/opt :origin) string?
   (ds/opt :amount_0) string?
   (ds/opt :amount_1) string?
   (ds/opt :amount_usd) string?
   (ds/opt :sqrt_price_x96) big-integer-spec
   (ds/opt :tick) big-integer-spec
   (ds/opt :log_index) big-integer-spec
   (ds/opt :evaluated_price) float?
   (ds/opt :evaluated_amount) float?
   (ds/opt :evaluated_aggressor) e-trade-aggressive-side-spec
   (ds/opt :pool_id) string?
   (ds/opt :transaction_id) string?
   })

(def swap-v3-dto-spec
  (ds/spec
    {:name ::swap-v3-dto
     :spec swap-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/SwapV3DTO.js
goog.provide('API.Client.SwapV3DTO');

/**
 * @record
 */
API.Client.SwapV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.SwapV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.SwapV3DTO.prototype.recvTime;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.SwapV3DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.SwapV3DTO.prototype.vid;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.id;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.transaction;

/**
 * 
 * @type {!Date}
 * @export
 */
API.Client.SwapV3DTO.prototype.timestamp;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.pool;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.token0;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.token1;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.sender;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.recipient;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.origin;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.amount0;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.amount1;

/**
 * 
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.amountUsd;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.SwapV3DTO.prototype.sqrtPriceX96;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.SwapV3DTO.prototype.tick;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.SwapV3DTO.prototype.logIndex;

/**
 * @type {!number}
 * @export
 */
API.Client.SwapV3DTO.prototype.evaluatedPrice;

/**
 * @type {!number}
 * @export
 */
API.Client.SwapV3DTO.prototype.evaluatedAmount;

/**
 * @type {!API.Client.ETradeAggressiveSide}
 * @export
 */
API.Client.SwapV3DTO.prototype.evaluatedAggressor;

/**
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.poolId;

/**
 * @type {!string}
 * @export
 */
API.Client.SwapV3DTO.prototype.transactionId;

